### ISSUE TITLE
libunwind does not reuse opened files

### ISSUE THREAD

**Issue by cmouse** [libunwind/libunwind#116]
When running a minimal reproduction code, we can see following behaviour with local tracing:

```
    open("/proc/18711/maps", O_RDONLY)      = 3
    open("/home/cmouse/src/unwind-stack/test", O_RDONLY) = 4
    close(4)                                = 0
    close(3)                                = 0
    #0 backtrace[0x556351635b40]
    open("/proc/18711/maps", O_RDONLY)      = 3
    open("/home/cmouse/src/unwind-stack/test", O_RDONLY) = 4
    close(4)                                = 0
    close(3)                                = 0
    #1 some_other_function[0x556351635d06]
    open("/proc/18711/maps", O_RDONLY)      = 3
    open("/home/cmouse/src/unwind-stack/test", O_RDONLY) = 4
    close(4)                                = 0
    close(3)                                = 0
    #2 some_function[0x556351635d17]
    open("/proc/18711/maps", O_RDONLY)      = 3
    open("/home/cmouse/src/unwind-stack/test", O_RDONLY) = 4
    close(4)                                = 0
    close(3)                                = 0
    #3 main[0x556351635d23]
    open("/proc/18711/maps", O_RDONLY)      = 3
    open("/lib/x86_64-linux-gnu/libc-2.24.so", O_RDONLY) = 4
    close(4)                                = 0
    close(3)                                = 0
    #4 __libc_start_main[0x7fc8d30431f0]
    open("/proc/18711/maps", O_RDONLY)      = 3
    open("/home/cmouse/src/unwind-stack/test", O_RDONLY) = 4
    close(4)                                = 0
    close(3)                                = 0
```

This shows the maps being opened & closed along with the binary for every symbol with no caching or reuse whatsoever. Perhaps it could at least reuse the last opened file and maps?

Minimal reproduction under #115

**Comment by djwatson** [comment]
Thanks for the detailed description.  I think this would be roughly fixed with #15 , which is what the original itanium unwinder did.

**Comment by bregma** [comment]
Closing as a duplicate of #15
